src/ServiceControl.Monitoring.Data/RawDataReporter.cs

```diff
@@ -67,17 +80,23 @@ public void Start()
                         await Task.Delay(singleSpinningTime).ConfigureAwait(false);
                     }
 
-                    await Consume().ConfigureAwait(false);
+                    if (consumers.Count >= ParallelConsumers)
+                    {
+                        var task = await Task.WhenAny(consumers).ConfigureAwait(false);
```

Comment:
@danielmarbach Could you validate my parallelism attempt? My reasoning was simple: either schedule another consume on the current thread, or if the set is saturated, await for any and use its thread (possibly IO) to schedule another one